packages/office-ui-fabric-react/src/utilities/DraggableZone/DraggableZone.tsx

```diff
@@ -0,0 +1,337 @@
+import * as React from 'react';
+import { IDraggableZoneProps, ICoordinates, IDragData } from './DraggableZone.types';
+import { getClassNames } from './DraggableZone.styles';
+import { EventGroup } from '../../Utilities';
+
+export interface IDraggableZoneState {
+  isDragging: boolean;
+  position: ICoordinates;
+  lastPosition?: ICoordinates;
+}
+
+const eventMapping = {
+  touch: {
+    start: 'touchstart',
+    move: 'touchmove',
+    stop: 'touchend'
+  },
+  mouse: {
+    start: 'mousedown',
+    move: 'mousemove',
+    stop: 'mouseup'
+  }
+};
+
+// These are needed so that we can generalize the events
+// and so we have access to clientX and clientY in the touch events
+type MouseTouchEvent<T> = React.MouseEvent<T> & React.TouchEvent<T>;
+
+export class DraggableZone extends React.PureComponent<IDraggableZoneProps, IDraggableZoneState> {
+  private _touchId?: number;
+  private _currentEventType = eventMapping.mouse;
+  private _events: EventGroup;
+
+  constructor(props: IDraggableZoneProps) {
+    super(props);
+
+    this.state = {
+      isDragging: false,
+      position: this.props.position || { x: 0, y: 0 },
+      lastPosition: undefined
+    };
+  }
+
+  public getDerivedStateFromProps(nextProps: IDraggableZoneProps) {
```

Comment:
`getDerivedStateFromProps` has to be static, which means you can't access `this` or the previous props. Also, unlike `componentWillReceiveProps`, I'm pretty sure it's called on every render, so you'll end up overwriting your position more often than you want.

For those reasons, I *think* the best place for this type of logic is `componentDidUpdate`, which has access to the previous props and state as well as current. (I say this tentatively because we're still developing patterns/best practices for moving away from `componentWillReceiveProps`.)

@dzearing FYI